gadgets/trace_tcpdrop/program.bpf.c

```diff
@@ -0,0 +1,191 @@
+// SPDX-License-Identifier: GPL-2.0
+//
+// Based on tcpdrop(8) from BCC
+//
+// Copyright 2018 Netflix, Inc.
+// 30-May-2018    Brendan Gregg   Created this.
+// 15-Jun-2022    Rong Tao        Add tracepoint:skb:kfree_skb
+// Copyright 2023 Microsoft Corporation
+
+#include <vmlinux.h>
+
+#include <bpf/bpf_helpers.h>
+#include <bpf/bpf_core_read.h>
+#include <bpf/bpf_tracing.h>
+#include <bpf/bpf_endian.h>
+
+#include <gadget/buffer.h>
+#include <gadget/types.h>
+#include <gadget/macros.h>
+
+#define GADGET_TYPE_TRACING
+#include <gadget/sockets-map.h>
+
+#define TASK_COMM_LEN 16
+
+// This enum is the same as the one in vmlinux.h, but redefined so that it can provide a name
+// to be used for `state` in struct event. This way we get a readable value for column `state`.
+enum tcp_state {
+	tcp_established = 1,
+	tcp_syn_sent = 2,
+	tcp_syn_recv = 3,
+	tcp_fin_wait1 = 4,
+	tcp_fin_wait2 = 5,
+	tcp_time_wait = 6,
+	tcp_close = 7,
+	tcp_close_wait = 8,
+	tcp_last_ack = 9,
+	tcp_listen = 10,
+	tcp_closing = 11,
+	tcp_new_syn_recv = 12,
+	tcp_max_states = 13,
+};
+
+struct event {
+	struct gadget_l4endpoint_t src;
+	struct gadget_l4endpoint_t dst;
+
+	gadget_timestamp timestamp;
+	enum tcp_state state;
+	__u8 tcpflags;
+	enum skb_drop_reason reason;
+	__u32 netns;
+
+	gadget_mntns_id mount_ns_id;
+	__u32 pid;
+	__u32 tid;
+	__u32 uid;
+	__u32 gid;
+	__u8 task[TASK_COMM_LEN];
```

Comment:
The original gadget has two instances of this fields:

https://github.com/inspektor-gadget/inspektor-gadget/blob/616a94c122f23e0372c3084e2bd77583a6b02960/pkg/gadgets/trace/tcpdrop/tracer/bpf/tcpdrop.h#L37-L38

I think it's fine to use only one of this for now given that we don't support sub-structures in the event. Would you mind adding a small comment indicating it should be converted to a structure once we support that?

Reply:
✅  Also added an entry to https://github.com/inspektor-gadget/inspektor-gadget/issues/2316#issue-2050087434